### ISSUE TITLE
Introduce `ctx.dateNow()`

### ISSUE THREAD

**Issue by slinkydeveloper** [restatedev/sdk-typescript#271]
Just a simple helper for 

```
const timestamp = await ctx.sideEffect(() => new Date().toJSON());
```

**Comment by marcus-sa** [comment]
That'd be misleading, because `ctx.dateNow()` would indicate a UNIX timestamp.
The implementation should use `Date.now()` not `new Date().toJSON()`.

**Comment by mupperton** [comment]
> That'd be misleading, because `ctx.dateNow()` would indicate a UNIX timestamp. The implementation should use `Date.now()` not `new Date().toJSON()`.

Makes sense - when I originally suggested this on Discord I was thinking more along the lines of having something like a `ctx.date.` with more than one method

So could do
`ctx.date.now() === Date.now()`
`ctx.date.toJSON() === new Date().toJSON()`

I think these are the most useful ones for now that can still be serialised in the journal

**Comment by marcus-sa** [comment]
> > That'd be misleading, because `ctx.dateNow()` would indicate a UNIX timestamp. The implementation should use `Date.now()` not `new Date().toJSON()`.
> 
> Makes sense - when I originally suggested this on Discord I was thinking more along the lines of having something like a `ctx.date.` with more than one method
> 
> So could do `ctx.date.now() === Date.now()` `ctx.date.toJSON() === new Date().toJSON()`
> 
> I think these are the most useful ones for now that can still be serialised in the journal

Yeah, that'd be a great API
